### ISSUE TITLE
custom-window-preview: issue collector

### ISSUE THREAD

**Issue by mweirauch** [home-sweet-gnome/dash-to-panel#657]
Fedora 29 with 3.30.2 and current custom-window-preview on Wayland at 200% scale.
Nothing fancy, except ungrouped windows.

I usuall don't use window previews due to gnome-shell getting even laggier with time then.
Gave the branch a try:
* in dash, I am not able to grab a window (from the middle) and drag it into the workspace previews on the right (to move it to another workspace)
* in dash, I am not able to grab a window from one workspace preview and move it to another workspace preview
* in dash - when one or more apps are open - I am unable to click the first two favourite icons in the left panel (there is also no hover effect); when I close the running apps, I can click the two topmost favourites again

So while this is basically dash-related, I think the new preview has an effect on this. Wen't back to current master and everything is fine again.

I also noticed some weird "no-mouse-click-registered" stuff, but I'd need to test further to give a reproducable example.

PS: I didn't want to hijack another thread. Is it ok to open up new issues or would you like to open a "collector-issue" for the new window preview?

**Comment by charlesg99** [comment]
Hello mweirauch! I don't have a gnome-shell 3.30 machine right now, but can't reproduce the dash issues so far on 3.22, 3.28 and 3.32 (wayland or xorg). Gonna have to install Fedora I guess :)

If you don't mind changing the title of your issue, we could keep the discussion about the new previews in here, since it's still a WIP on a feature branch. Once it is merged to master, then regular issues will do just fine.

Thanks!

**Comment by mweirauch** [comment]
Could reproduce the my described "drag&move" issues on a 99.9% idencital system (same hw, sw, only xorg). The mouse-click-issue on the favourite-icons in the dash (not DtP) couldn't be reproduced. Do you fear some other extension interfering? The only one I can think of is "Launch new instance", but this one shouldn't interfere when dragging application windows with the dash.

**Comment by charlesg99** [comment]
> The only one I can think of is "Launch new instance", but this one shouldn't interfere when dragging application windows with the dash.

I don't see how the new previews could interfere with the dash's dnd either :) Not saying they aren't, I just can't think of why that would be. I'll try to reproduce this some more! Thanks!

**Comment by mweirauch** [comment]
A little update - same env as above, but this time with unscaled (100%) UI. (Disabled all other extensions during testing.)

I got a fixed 4 workspace layout for GS: Dragging - in dash - from ws1 to ws2 or vice versa doesn't work. But dragging from ws1 or ws2 to ws3 or ws4 works. Dragging from ws3 to ws4, or ws4 to ws3 works, but dragging from or to ws1 or ws2 doesn't.
Judging from a visual perspective - in dash - the workspace preview for ws2 seems to end at the same height as the window previews.

Now I dunno how the previews are implemented technically, but perhaps there's some hidden area/overlay which is still there "stealing the focus" from the underlying UI elements?

I couldn't find a way to reproduce yet, but I mentioned this "no-mous-click-registered"-issue in the original post and I had it again, where in Chrome (as an example) I couldn't click the Github icon or any other link in the top area of the page after it having focus again.

No biggi for me, as I am not using previews (anymore), but just wanted to mention.

**Comment by charlesg99** [comment]
Thank you mweirauch, you deduced right! I could reproduce the problem (with my panel on top :) and it was indeed a weird "overview only" behavior caused by the preview container being over the workspace thumbnails! This should now be fixed if you want to confirm that it works for you as well. Thanks!

**Comment by mweirauch** [comment]
Confirmed working! Nice I could give back a bit helping you guys develop this nice extension!

**Comment by ubluish** [comment]
Thanks for such amazing work, In Windows preview options window there is no scrollbar so i cant see the bottom of the page.
![Untitled](https://user-images.githubusercontent.com/21042432/58742132-0fc78800-8422-11e9-9c57-ab610b0ab77e.png)

**Comment by charlesg99** [comment]
Hello ubluish, a scrollbar was added to the options dialog. Could you please pull/install again? Thanks!

**Comment by ubluish** [comment]
Hello @charlesg99, i tested it and everything is working fine. Thanks!

**Comment by digiberk** [comment]
Thanks for the amazing work guys. I've pulled this branch down a day ago and the window previews are stunning. Also there seems to be less mouse lag under Wayland (Intel graphics). I can't wait for this to hit the master branch!

**Comment by charlesg99** [comment]
Hello, please pull/install again as I'm now done with the development. Should be merge time soon! Thanks!

**Comment by charlesg99** [comment]
Hey guys, thank you for the feedback! The new window previews are now on master, so please open a new issue if something else comes up! Thanks!
